PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/layers/layers_wrapper.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include "transaction_layer.h"
+#include "datalink.h"
+#include "../utils/tlp.h"
+#include "../utils/queue_wrapper.h"
+#include <queue>
+
+class LayersWrapper
+{
+public:
+	TransactionLayer* transaction;
+	DatalinkLayer* datalink;
+	TLP* partialTlp;
+
+	/**
+	 * @brief Sends NOP Flit to the specified queue
+	 * This function sends NOP Flits to the specified queue for each credit type.
+	 * @param P_CREDIT an array of two integers representing the available P credits
+	 * @param NP_CREDIT an array of two integers representing the available NP credits
+	 * @param CPL_CREDIT an array of two integers representing the available CPL credits
+	 * @param dllpType the DLLP type for the NOP Flits
+	 * @param sendOn a pointer to a QueueWrapper object where the NOP Flit queue will be pushed
+	 * @return void
+	*/
+	void sendNOPFlit(int P_CREDIT[2], int NP_CREDIT[2], int CPL_CREDIT[2], Dllp::DllpType dllpType, QueueWrapper<std::queue<Flit*>*>* sendOn);
```

Comment:
As per yesterday's meeting, the queue wrapper now will hold Flit* instead of std::queue<Flit*>. It is also suggested to take the globals object directly as in the class diagram instead of taking all the arrays.
One last thing: the arrays names were changed to hold shared and dedicated credits, so it's always recommended to rebase on main before pushing to keep track of latest changes